aiaccel/job/dispatcher.py

```diff
@@ -0,0 +1,339 @@
+        y = self.collect_result()
+        try:
+            y = ast.literal_eval(y)
+        except ValueError:
+            y = y
+        self.result = y
+        return self.result
+
+    def get_job_numbers(self, job_name: str) -> list[int]:
+        """
+        Get the job numbers of the same job name.
+        """
+        cmd = "qstat -xml"
+        data = subprocess.run(cmd.split(), capture_output=True, text=True)
+        qstat_dict = xmltodict(data.stdout)
+        if qstat_dict["queue_info"] == "":
+            return []
+        job_numbers = []
+        for queue_info in qstat_dict["queue_info"]:
+            for job_list in queue_info["job_list"]:
+                if job_list["JB_name"] == job_name:
+                    job_numbers.append(int(job_list["JB_job_number"]))
+        return job_numbers
+
+    @retry(_MAX_NUM=60, _DELAY=3.0)
+    def _update_job_number(self) -> None:
+        jn = self.get_job_numbers(self.job_name)
+        for job_number in jn:
+            if job_number not in self._seen_job_numbers:
+                self._job_number = job_number
+                self._seen_job_numbers.add(job_number)
+                break
+        if self._job_number is None:
+            raise ValueError(f"Job number not found: {self.job_name}")
+
+    def _cleanup_output_files(self) -> None:
+        if isinstance(self.stdout_file_path, Path) and self.stdout_file_path.exists():
+            self.stdout_file_path.unlink()
+        if isinstance(self.stderr_file_path, Path) and self.stderr_file_path.exists():
+            self.stderr_file_path.unlink()
+
+    def _execute_command(self) -> None:
+        print(f"{self.cmd}")
+        subprocess.run(self.cmd.split(), capture_output=True, text=True)
+
+    def _update_output_files_path(self) -> None:
+        """
+        -o [stdout_file_path] -e [stderr_file_path] による指定がない場合，
+        ジョブ名(バッチファイル名)に応じたファイル名を生成する
+        """
+        if self.stdout_file_path is None or self.stderr_file_path is None:
+            if self._job_number is None:
+                self._update_job_number()
+            base_path = Path("./").resolve()
+            if self.stdout_file_path is None:
+                self.stdout_file_path = (
+                    base_path / f"{self.job_name}.o{self._job_number}"
+                )
+            if self.stderr_file_path is None:
+                self.stderr_file_path = (
+                    base_path / f"{self.job_name}.e{self._job_number}"
+                )
+
+    def _wait_for_job_submit_confirm(self) -> None:
+        try:
+            self._update_job_number()  # wait for the job to be submitted
+            self.update_state()
+            if self.get_state() in ["r", "s", "q", "w"]:
+                self._submitted = True
+            if (
+                self.stdout_file_path is not None and self.stdout_file_path.exists()
+            ) or (self.stderr_file_path is not None and self.stderr_file_path.exists()):
+                self._submitted = True
```

Comment:
```suggestion
            if self.stdout_file_path.exists() or self.stderr_file_path.exists():
                self._submitted = True
```
↑に関連して，型チェックを減らしたいですね．
ただ，この部分って，"r"だと判定できるだけで良いのではと思ったのですが，如何でしょうか．